### PR TITLE
Vision Suite refactor

### PR DESCRIPTION
**Pull request by JKL98ISR**
Vision Suite refactor + some renaming and general fixes

### CODE REVIEW

**Review by ItayGabbay — APPROVED** [review]

File: deepchecks/vision/base.py [hunk 1]
```diff
@@ -409,6 +396,13 @@ def run(
                 run_train_test_checks=run_train_test_checks,
                 results=results
             )
+            for check_idx, check in checks.items():
+                if check_idx not in results:
```
Comment (matanper) [inline deepchecks/vision/base.py:400]: why would check_idx be in results?

File: deepchecks/vision/base.py [hunk 2]
```diff
@@ -409,6 +396,13 @@ def run(
...
+                    if str(check_idx).endswith('Train'):
```
Comment (matanper) [inline deepchecks/vision/base.py:401]: IMO to compare strings is not ideal, I'd do 3 dicts (checks, single_test, single_train)

File: deepchecks/vision/base.py (hunk 1 above)
Comment (JKL98ISR) [reply deepchecks/vision/base.py:400]: if there was exception

File: deepchecks/vision/base.py [hunk 3]
```diff
@@ -374,25 +373,13 @@ def run(
 
         for check_idx, check in list(self.checks.items()):
             if isinstance(check, (TrainTestCheck, ModelOnlyCheck)):
-                checks[check_idx] = copy.deepcopy(check)
+                check.initialize_run(context)
+                checks[check_idx] = check
             elif isinstance(check, SingleDatasetCheck):
                 if train_dataset is not None:
-                    checks[str(check_idx) + ' - Train'] = copy.deepcopy(check)
+                    checks[str(check_idx) + ' - Train'] = check
```
[the remark this answers is not in the document]
Comment (JKL98ISR) [reply deepchecks/vision/base.py:380]: this is why I did the other changes.
so init and compute are concurrent for each

**Review by matanper — APPROVED** [review]

**Pull request merged**